### PR TITLE
INTMDB-489 project invitation

### PR DESCRIPTION
**Pull request by ParthasarathyVaradhan**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ✅ } Bug fix (non-breaking change which fixes an issue)
- [ ✅  ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ✅  ] This change requires a documentation update

## Required Checklist:

- [ ✅ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ✅  ] I have added tests that prove my fix is effective or that my feature works
- [ ✅  ] I have added any necessary documentation (if appropriate)
- [✅  ] I have run `make fmt` and formatted my code

## Further comments

Examples aws cloudformation template is available in ./examples/project-invitation/ folder.

### CODE REVIEW

**Review by matt-condon — APPROVED** [review]
non-blocking comments so adding LGTM

File: cfn-resources/project-invitation/cmd/resource/resource.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+		Roles:    currentModel.Roles,
+		Username: *currentModel.Username,
+	}
+
+	invitation, res, err := client.Projects.InviteUser(context.Background(), *currentModel.ProjectId, invitationReq)
+	if err != nil {
+		_, _ = log.Warnf("Create - error: %+v", err)
+		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+	currentModel.Id = &invitation.ID
```
[the remark this answers is not in the document]
Comment (ParthasarathyVaradhan) [reply cfn-resources/project-invitation/cmd/resource/resource.go:60]: We are following Camel case standards "Id",
"invitation" is object returned from Atlas Go SDK (&invitation.ID).

`
// https://github.com/mongodb/go-client-mongodb-atlas/blob/master/mongodbatlas/organization_invitations.go
// Invitation represents the structure of an Invitation.
type Invitation struct {
	ID              string   `json:"id,omitempty"`
	OrgID           string   `json:"orgId,omitempty"`
	OrgName         string   `json:"orgName,omitempty"`
	GroupID         string   `json:"groupId,omitempty"`
	GroupName       string   `json:"groupName,omitempty"`
	CreatedAt       string   `json:"createdAt,omitempty"`
	ExpiresAt       string   `json:"expiresAt,omitempty"`
	InviterUsername string   `json:"inviterUsername,omitempty"`
	Username        string   `json:"username,omitempty"`
	Roles           []string `json:"roles,omitempty"`
	TeamIDs         []string `json:"teamIds,omitempty"`
}`

We might have to create ticket with the respective team for this change.

File: cfn-resources/project-invitation/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+export AWS_SECRET_ACCESS_KEY=""
+export AWS_REGION=""
+export AWS_DEFAULT_REGION=""
+
+#Command to deploy the sample Project-Invitation stack (Before this step "cfn submit" should have been executed successfully)
+./examples/project-invitation/deploy.sh
+```
+
+
+For more information see: MongoDB Atlas API Project Invitation [Endpoint](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Projects/operation/returnAllProjectInvitations) Documentation.
```
Comment (Zuhairahmed) [inline cfn-resources/project-invitation/README.md:61]:
```suggestion
For more information see: MongoDB Atlas API [Project Invitation Endpoint](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Projects/operation/returnAllProjectInvitations) Documentation.
```

File: cfn-resources/project-invitation/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+## Installation
+```
+TAGS=logging make
+cfn submit --verbose --set-default
+```
+
+## Usage
+
+Examples aws cloudformation template is available here [example template.](../../examples/project-invitation/project-invitation.json)
```
Comment (Zuhairahmed) [inline cfn-resources/project-invitation/README.md:46]:
```suggestion
Example AWS CloudFormation template is available here [example template.](../../examples/project-invitation/project-invitation.json)
```

File: cfn-resources/project-invitation/README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+Both CREATE & DELETE tests must pass.
+
+## Installation
+```
+TAGS=logging make
+cfn submit --verbose --set-default
+```
+
+## Usage
```
Comment (Zuhairahmed) [inline cfn-resources/project-invitation/README.md:44]:
```suggestion
## Example and Usage 
```

**Review by Zuhairahmed — APPROVED** [review]
Nice work @ParthasarathyVaradhan! Few minor comments below if you can adjust, not blocking.

File: cfn-resources/project-invitation/Makefile [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ldXflagsD=github.com/mongodb/mongodbatlas-cloudformation-resources/util.defaultLogLevel=debug
+build:
+	#make -f makebuild  # this runs build steps required by the cfn cli
+	cfn generate
+	env GOOS=$(goos) CGO_ENABLED=$(cgo) GOARCH=$(goarch) go build -ldflags="-s -w -X '$(ldXflags)'" -tags="$(tags)" -o bin/handler cmd/main.go
+
+debug:
+	#make -f makebuild  # this runs build steps required by the cfn cli
+	cfn generate
+	env GOOS=$(goos) CGO_ENABLED=$(cgo) GOARCH=$(goarch) go build -ldflags="-s -w -X '$(ldXflagsD)'" -tags="$(tags)" -o bin/handler cmd/main.go
```
Comment (Adelmar92) [inline cfn-resources/project-invitation/Makefile:20]:
```suggestion
	cfn generate
	env GOOS=$(goos) CGO_ENABLED=$(cgo) GOARCH=$(goarch) go build -ldflags="-s -w -X '$(ldXflags)'" -tags="$(tags)" -o bin/handler cmd/main.go

debug:
	cfn generate
	env GOOS=$(goos) CGO_ENABLED=$(cgo) GOARCH=$(goarch) go build -ldflags="-s -w -X '$(ldXflagsD)'" -tags="$(tags)" -o bin/handler cmd/main.go
```

File: cfn-resources/project-invitation/Makefile [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+
+clean:
+	rm -rf bin
```
Comment (Adelmar92) [inline cfn-resources/project-invitation/Makefile:24]:
```suggestion
	rm -rf bin
	
```

File: cfn-resources/project-invitation/cmd/resource/model.go [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+// Model is autogenerated from the json schema
+type Model struct {
+	ApiKeys         *ApiKeyDefinition `json:",omitempty"`
+	CreatedAt       *string           `json:",omitempty"`
+	ExpiresAt       *string           `json:",omitempty"`
+	ProjectId       *string           `json:",omitempty"`
+	Id              *string           `json:",omitempty"`
+	TotalCount      *float64          `json:",omitempty"`
+	PageNum         *int              `json:",omitempty"`
```
Comment (Adelmar92) [inline cfn-resources/project-invitation/cmd/resource/model.go:13]: total count and page num is used for the listing, but i dont see we are using it, or should we use it to make an efficient listing?

File: cfn-resources/project-invitation/cmd/resource/resource.go [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+}
+
+func setup() {
+	util.SetupLogger("mongodb-atlas-project-invitation")
+}
+
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	_, _ = log.Debugf("Create() currentModel:%+v", currentModel)
```
Comment (Adelmar92) [inline cfn-resources/project-invitation/cmd/resource/resource.go:34]:
```suggestion
	 log.Debugf("Create() currentModel:%+v", currentModel)
```

Comment (Adelmar92) [reply cfn-resources/project-invitation/cmd/resource/resource.go:34]: the same in all the others

Comment (ParthasarathyVaradhan) [reply cfn-resources/project-invitation/cmd/resource/resource.go:34]: on implementing this suggested change, our lint checks fails.

Comment (Adelmar92) [reply cfn-resources/project-invitation/cmd/resource/resource.go:34]: okok we need to see in the future if we can delete the return of those logs, if we are not going to use them

**Review by Adelmar92 — APPROVED** [review]

**Pull request merged**